### PR TITLE
chore: automatically generate dbauthz when new queries are added

### PR DESCRIPTION
**Pull request by kylecarbs**

### CODE REVIEW

**Review by Emyrk — CHANGES_REQUESTED** [review]
Need to get the system.go funcs

File: coderd/database/dbauthz/dbauthz_test.go [hunk 1]
```diff
@@ -155,3 +158,1143 @@ func must[T any](value T, err error) T {
...
+		}).Asserts(key, rbac.ActionRead).Returns(key)
+	}))
+	s.Run("GetAPIKeysByLoginType", s.Subtest(func(db database.Store, check *expects) {
+		a, _ := dbgen.APIKey(s.T(), db, database.APIKey{LoginType: database.LoginTypePassword})
+		b, _ := dbgen.APIKey(s.T(), db, database.APIKey{LoginType: database.LoginTypePassword})
+		_, _ = dbgen.APIKey(s.T(), db, database.APIKey{LoginType: database.LoginTypeGithub})
+		check.Args(database.LoginTypePassword).
+			Asserts(a, rbac.ActionRead, b, rbac.ActionRead).
+			Returns(slice.New(a, b))
+	}))
```
Comment (Emyrk) [inline coderd/database/dbauthz/dbauthz_test.go:188]: Would be really nice to generate these too. Your call on how difficult that is though.

Maybe I could make the test failure just print out the test to copy/paste. :thinking: 

happy to brainstorm this

File: coderd/database/dbauthz/dbauthz.go [hunk 2]
```diff
@@ -456,3 +460,1845 @@ func prepareSQLFilter(ctx context.Context, authorizer rbac.Authorizer, action rb
...
+	// filter would slow down the query for no benefit.
+	if err := q.authorizeContext(ctx, rbac.ActionRead, rbac.ResourceAuditLog); err != nil {
+		return nil, err
+	}
+	return q.db.GetAuditLogsOffset(ctx, arg)
+}
+
+func (q *querier) GetAuthorizationUserRoles(ctx context.Context, userID uuid.UUID) (database.GetAuthorizationUserRolesRow, error) {
+	panic("Not implemented")
+}
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/dbauthz/dbauthz.go:792]: I think it'd be simpler if we merged them all, just like we do for the other generated files. What do you think?

File: coderd/database/dbauthz/dbauthz_test.go (hunk 1 above)
Comment (kylecarbs) [reply coderd/database/dbauthz/dbauthz_test.go:188]: I think we can do that post-merge or something... right now it's not a massive pain-point for me, so I think we can punt it.

File: coderd/database/dbauthz/dbauthz.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Emyrk) [reply coderd/database/dbauthz/dbauthz.go]: We can... It was a nice organizational tool when we were figuring it out. Going forward though we handle things on a case by case basis, so I think the difference in files is less important.

**Review by Emyrk — APPROVED** [review]

**Pull request merged**